File: ej2.cpp
#include <iostream>
#include <vector>
using namespace std;
class Arreglo {
private:
    vector<string> Nombres;

public:
    void AgregarNombre(string nombre) {
        Nombres.push_back(nombre);
    }

    string ObtenerNombre(int posicion) {
        if (posicion >= 0 && posicion < Nombres.size())
            return Nombres[posicion];
        else
            return "Posicion invalida";
    }

    int Cantidad() {
        return Nombres.size();
    }
};

int main() {
    Arreglo Arr; 

    string Nombre;
    char Continuar;

    do {
        cout << "Ingrese su nombre: ";
        cin >> Nombre;
        Arr.AgregarNombre(Nombre);

        int UltimaPos = Arr.Cantidad() - 1;

        cout << "El nombre guardado en la posicion " << UltimaPos
             << " es: " << Arr.ObtenerNombre(UltimaPos) << endl;

        cout << "Desea continuar? (S/N): ";
        cin >> Continuar;

    } while (Continuar == 'S');

    return 0;
}

File: ej2.py
class Arreglo:
    def __init__(self):
        self.nombres = []

    def agregar_nombre(self, nombre):
        self.nombres.append(nombre)

    def obtener_nombre(self, posicion):
        if 0 <= posicion < len(self.nombres):
            return self.nombres[posicion]
        else:
            return "Posición inválida"

    def cantidad(self):
        return len(self.nombres)


def main():
    arr = Arreglo()
    while True:
        nombre = input("Ingrese su nombre: ")
        arr.agregar_nombre(nombre)

        ultima_pos = arr.cantidad() - 1
        print(f"El nombre guardado en la posición {ultima_pos} es: {arr.obtener_nombre(ultima_pos)}")

        continuar = input("Desea continuar? (S/N): ").upper()
        if continuar != 'S':
            break


if __name__ == "__main__":
    main()